source/tyr/tyr/resources.py

```diff
@@ -1691,43 +1694,78 @@ def check_db():
         return None
 
 
+@marshal_with(job_fields)
+def check_cities_job():
+    """
+    Check status of cities job in Tyr db
+    :return: the latest cities job
+    """
+    return (
+        models.Job.query.join(models.DataSet)
+        .filter(models.DataSet.type == 'cities')
+        .order_by(models.Job.created_at.desc())
+        .first()
+    )
+
+
 class CitiesStatus(flask_restful.Resource):
     def get(self):
+        response = {}
         if not current_app.config['CITIES_DATABASE_URI']:
             return {'message': 'cities db not configured'}, 404
-        msg = check_db()
-        if msg:
-            return {'message': 'cities db alembic version = {}'.format(msg)}, 200
+        cities_version = check_cities_db()
+        if cities_version:
+            response['cities db version'] = '{}'.format(cities_version)
         else:
             return {'message': 'cities db not reachable'}, 404
 
+        cities_job = check_cities_job()
+        if cities_job and 'instance' in cities_job:
+            # No instance associated to 'cities' job, remove the item
+            cities_job.pop('instance', None)
+            response['latest_job'] = cities_job
+
+        return response, 200
+
 
 class Cities(flask_restful.Resource):
     def post(self):
-        if not check_db():
+        if not check_cities_db():
             return {'message': 'cities db not reachable'}, 404
 
         parser = reqparse.RequestParser()
         parser.add_argument('file', type=werkzeug.FileStorage, location='files')
         args = parser.parse_args()
 
         if not args['file']:
-            logging.info("No file provided")
+            logging.error("No file provided")
             return {'message': 'No file provided'}, 400
 
         f = args['file']
         file_name = f.filename
         file_path = str(os.path.join(os.path.abspath(current_app.config['CITIES_OSM_FILE_PATH']), file_name))
         f.save(file_path)
-
-        logging.info("file: {}".format(f))
+        logging.info("file received: {}".format(f))
+
+        # Create Job and Dataset in db to track progress
+        job = models.Job()
+        job.state = 'running'
+        models.db.session.add(job)
+        dataset = models.DataSet()
+        dataset.name = file_path
+        dataset.family_type = 'cities'
+        models.db.session.add(dataset)
+        job.data_sets.append(dataset)
 
         if COSMOGONY_REGEXP.match(file_name):
             # it's a cosmogony file, we import it with cosmogony2cities
-            cosmogony2cities.delay(file_path)
+            dataset.type = exe = 'cosmogony2cities'
         else:
             # we import it the 'old' way, with cities
-            cities.delay(file_path)
+            dataset.type = exe = 'cities'
+
+        models.db.session.commit()
+        cities.delay(file_path, job.id, exe)
```

Comment:
can you make a try catch? because cities.delay may throw exceptions(ex. rabbitmq is down)
```python
try:
    cities.delay(file_path, job.id, exe)
except ... :
    job.state = 'fail'
    models.db.session.commit()

```